lib/shared/config-validator.js

```diff
@@ -183,8 +184,9 @@ function validateRules(
 
     Object.keys(rulesConfig).forEach(id => {
         const rule = getAdditionalRule(id) || BuiltInRules.get(id) || null;
+        const clonedRulesConfigForId = _.cloneDeep(rulesConfig[id]);
```

Comment:
Sorry to keep coming back to this, but what is the limitation that is stopping us from cloning the contents from any configuration files read when they're read, rather than only doing this for rule validation? Apologies if I'm just missing something here, and appreciate your patience :)

Reply:
I did tried the implementation earlier and it was not fixing the issue as it was sending the cloned config forward. 

https://github.com/eslint/eslint/pull/13034#discussion_r415295775.